auth.go

```diff
@@ -160,22 +178,41 @@ func (s *slackAuth) runServer() error {
 }
 
 func (s *slackAuth) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	code := r.FormValue("code")
-	resp, err := s.api.GetOAuthResponse(s.clientID, s.clientSecret, code, s.debug)
-	if err != nil {
-		log15.Error("error getting oauth response", "err", err.Error())
-		if err := s.errorTpl.Execute(w, resp); err != nil {
-			log15.Error("error displaying error tpl", "err", err.Error())
-		}
+	path := getPath(r)
+	if isIgnoredUri(path) {
+		return
+	}
+
+	log15.Info("requested: " + path)
+	handler, ok := s.httpHandlers[path]
+	if !ok {
+		w.WriteHeader(404)
+		log15.Error("no handler found", "err", path)
 		return
 	}
+	handler(w, r)
+}
+
+func (s *slackAuth) getAuthorizationHandler() func(w http.ResponseWriter, r *http.Request) {
```

Comment:
if all the function does is return a function and captures no parameter, why return a function and not be the function itself?